homeassistant/components/camera/local_file.py

```diff
@@ -11,31 +11,44 @@
 import voluptuous as vol
 
 from homeassistant.const import CONF_NAME
-from homeassistant.components.camera import Camera, PLATFORM_SCHEMA
+from homeassistant.components.camera import (
+    Camera, CAMERA_SERVICE_SCHEMA, DOMAIN, PLATFORM_SCHEMA)
 from homeassistant.helpers import config_validation as cv
 
 _LOGGER = logging.getLogger(__name__)
 
 CONF_FILE_PATH = 'file_path'
-
 DEFAULT_NAME = 'Local File'
+SERVICE_UPDATE_FILE_PATH = 'local_file_update_file_path'
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
     vol.Required(CONF_FILE_PATH): cv.string,
     vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string
 })
 
+CAMERA_SERVICE_UPDATE_FILE_PATH = CAMERA_SERVICE_SCHEMA.extend({
+    vol.Required(CONF_FILE_PATH): cv.string
+})
+
 
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Set up the Camera that works with local files."""
     file_path = config[CONF_FILE_PATH]
+    camera = LocalFile(config[CONF_NAME], file_path)
 
-    # check filepath given is readable
-    if not os.access(file_path, os.R_OK):
-        _LOGGER.warning("Could not read camera %s image from file: %s",
-                        config[CONF_NAME], file_path)
+    def update_file_path_service(call):
+        """Update the file path."""
+        file_path = call.data.get(CONF_FILE_PATH)
+        camera.update_file_path(file_path)
+        return True
 
-    add_devices([LocalFile(config[CONF_NAME], file_path)])
+    hass.services.async_register(
```

Comment:
The platform is not async (maybe you could change and make it async) so you should not use **async_register** but **register**

Reply:
OK